### PR TITLE
Color val delete

### PR DESCRIPTION
**Pull request by minggo**
Re: #13229

### Changelog

* Remove Color._val, use r, g, b, a to record each value.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/math/color.ts [hunk 1]
```diff
@@ -118,10 +117,10 @@ export class Color extends ValueType {
      */
     public static fromVec4 (value: Vec4, out?: Color): Color {
         out = out === undefined ? new Color() : out;
-        out.r = Math.floor(value.x / toFloat);
-        out.g = Math.floor(value.y / toFloat);
-        out.b = Math.floor(value.z / toFloat);
-        out.a = Math.floor(value.w / toFloat);
+        out.r = value.x / toFloat;
```
Comment (dumganhar) [inline cocos/core/math/color.ts:120]: Does't need `Math.floor` any more here?

Comment (minggo) [reply cocos/core/math/color.ts:120]: The setter will handle it.

File: cocos/core/math/color.ts [hunk 2]
```diff
@@ -214,6 +212,10 @@ export class Color extends ValueType {
      * @zh 全通道统一缩放颜色
      */
     public static scale<Out extends IColorLike> (out: Out, a: Out, b: number): Out {
+        if (b < 0) {
+            return out;
```
Comment (dumganhar) [inline cocos/core/math/color.ts]: Why add the `if (b < 0)` check here?

Comment (minggo) [reply cocos/core/math/color.ts]: Scale a negative value is meaningless.

File: cocos/core/math/color.ts [hunk 3]
```diff
@@ -734,43 +735,10 @@ export class Color extends ValueType {
...
-
-    /**
-     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
-     */
-    public _set_a_unsafe (alpha: number): Color {
-        this._val = ((this._val & 0x00ffffff) | (alpha << 24)) >>> 0;
+        this.r *= other._r;
+        this.g *= other._g;
+        this.b *= other._b;
+        this.a = other._a;
```
Comment (dumganhar) [inline cocos/core/math/color.ts]: Why doesn't `a` need to be `multiplied`.

Comment (minggo) [reply cocos/core/math/color.ts]: fixed

File: cocos/core/math/color.ts (hunk 2 above)
Comment (minggo) [reply cocos/core/math/color.ts]: revert it

**Review by dumganhar — APPROVED** [review]

**Pull request merged**